Tests/Fuzzers/FuzzPacketParse.cpp

```diff
@@ -0,0 +1,24 @@
+#include <iostream>
+#include <IPv4Layer.h>
+#include <IPv6Layer.h>
+#include <unistd.h>
+#include <Packet.h>
+#include <PcapFileDevice.h>
+#include <RawPacket.h>
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *Data, size_t Size)
+{
+    struct timespec ts;
+    timespec_get(&ts, TIME_UTC);
+
+	if (Size < 2)
+		return 0;
+
+	// read the first (and only) packet from the file
+	pcpp::RawPacket rawPacket(((const uint8_t *)Data + 1), Size - 1, ts, false, pcpp::LinkLayerType(Data[0]));
+	// parse the raw packet into a parsed packet
+	pcpp::Packet parsedPacket(&rawPacket);
+	// Convert the parsed packet to string to trigger all the parsing formats
+	parsedPacket.toString();
```

Comment:
I'm not sure we really need to trigger `toString()` (although we do it now). Creating a `pcpp::Packet` out of a `pcpp::RawPacket` already does all the parsing:

It calls `setRawPacket()`: https://github.com/seladb/PcapPlusPlus/blob/1d29c00f441dc50650b500bb6b6fe72efce28990/Packet%2B%2B/src/Packet.cpp#L127

Which in turn creates all the layers: https://github.com/seladb/PcapPlusPlus/blob/1d29c00f441dc50650b500bb6b6fe72efce28990/Packet%2B%2B/src/Packet.cpp#L58

Reply:
I saw the toString functions call a bunch of functions that aren't called otherwise. Seemed to be the most straightforward way to improve coverage. I will double check

Reply:
yes, in some protocols `toString()` does call a bunch of other functions (and in some not), so it really depends on how much of the PcapPlusPlus API we want to invoke with fuzzing testing. Probably the more the better 😃 but we should think what makes sense...